pydicom/values.py

```diff
@@ -379,20 +379,18 @@ def convert_value(VR, raw_data_element, encoding=default_encoding):
         logger.debug('unable to translate tag %s with VR %s'
                      % (raw_data_element.tag, VR))
 
-        for convert_vr in convert_retry_VR_order:
-            vr = convert_vr
-            converter = converters[vr]
+        for vr in convert_retry_VR_order:
             if vr == VR:
                 continue
             try:
                 value = convert_value(vr, raw_data_element, encoding)
-                break
-            except Exception:
-                pass
-            else:
```

Comment:
This is where the problem is.  The else clause is meant to be with the for loop rather than the try.

If the for loop exhausts all the attempts to convert the raw data to a useful representation, then the logger statement should say "Could not convert this value with any VR (in the convert_retry_VR_order list)". It should then just copy the raw value to the returned value (since this section is only ever invoked if the user has configured pydicom to not raise exceptions on invalid VR values.